tests/test_core.py

```diff
@@ -1859,94 +1869,80 @@ class TestO:
-        self.assertIs(obj, res)
-
-        # Check that we no longer need to release it.
-        self.assertFalse(obj._needs_release)
-
-        # Delete it and make sure that we don't segfault on garbage collection.
-        del obj
-        gc.collect()
-
-    def test_objcinstance_retain_release(self):
-        NSString = ObjCClass("NSString")
-
-        # Create an object which we don't own.
-        string = NSString.stringWithString("test")
-
-        # Check that it is not marked for release.
-        self.assertFalse(string._needs_release)
-
-        # Explicitly retain the object.
-        res = string.retain()
-
-        # Check that autorelease call returned the object itself.
-        self.assertIs(string, res)
+    def test_objcinstance_created_gc_released(self):
+        # Create an object which we own.
+        obj = NSObject.alloc().init()
 
-        # Manually release the object.
-        string.release()
+        wr = ObjcWeakref.alloc().init()
+        wr.weak_property = obj
 
         # Delete it and make sure that we don't segfault on garbage collection.
-        del string
-        gc.collect()
-
-    def test_objcinstance_dealloc(self):
-        class DeallocTester(NSObject):
-            attr0 = objc_property()
-            attr1 = objc_property(weak=True)
-
-            @objc_method
-            def dealloc(self):
-                self._did_dealloc = True
+        with autoreleasepool():
+            del obj
+            gc.collect()
 
-        obj = DeallocTester.alloc().init()
-        obj.__dict__["_did_dealloc"] = False
+        # Assert that the obj was deallocated.
+        self.assertIsNone(wr.weak_property)
 
-        attr0 = NSObject.alloc().init()
-        attr1 = NSObject.alloc().init()
+    def test_objcinstance_explicit_retained_gc_released(self):
+        with autoreleasepool():
+            # Create an object which we don't own.
+            NSString = ObjCClass("NSString")
+            obj = NSString.stringWithString(str(uuid.uuid4()))
 
-        obj.attr0 = attr0
-        obj.attr1 = attr1
+            wr = ObjcWeakref.alloc().init()
+            wr.weak_property = obj
 
-        self.assertEqual(attr0.retainCount(), 2)
-        self.assertEqual(attr1.retainCount(), 1)
+            # Delete it and make sure that we don't segfault on garbage collection.
+            del obj
+            gc.collect()
 
-        # ObjC object will be deallocated, can only access Python attributes afterwards.
-        obj.release()
-
-        self.assertTrue(obj._did_dealloc, "custom dealloc did not run")
-        self.assertEqual(
-            attr0.retainCount(), 1, "strong property value was not released"
-        )
-        self.assertEqual(attr1.retainCount(), 1, "weak property value was released")
+        # Assert that the obj was deallocated.
+        self.assertIsNone(wr.weak_property)
+
+    # def test_objcinstance_dealloc(self):
+    #     class DeallocTester(NSObject):
+    #         attr0 = objc_property()
```

Comment:
Commented-out test should be cleaned up.

Reply:
Darn, I forgot about this one. I did not want to remove this test, just update it. The  old "method of testing" now causes a segfault at the end. Updated now.